### ISSUE TITLE
Showcase featured images?

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#324]
Just a wild idea, looking forward to your feedback about it:
How about putting a "List of recently featured images" at the top of the gallery?

Plus:
- Show users what a great Commons picture is
- Inspire/motivate users to take part in the same project
- Let users hope that one day their picture will be shown at that place (in fact, show them that taking great pictures is worth the effort)

Minus:
- Quite a stretch from the core mission of the app
- Potentially time wasting
- Certainly bandwidth wasting

**Comment by misaochan** [comment]
I like the idea of having a list of recently featured images.

Personally, though,I wouldn't really want it at the top of the gallery. After I upload a pic, I like to glance at the top pic of the gallery to check that the upload has completed, etc. I also like to be able to quickly check what the last couple of photos I uploaded was, without having to scroll.

Perhaps we could just have it as an additional activity, so users can decide whether to view it or not?

**Comment by tobias47n9e** [comment]
I would like a gallery of featured pictures, if it is a separate item in the nav drawer (https://material.io/guidelines/patterns/navigation-drawer.html). That way it only loaded if a user specifically wants to look at the gallery.

**Comment by misaochan** [comment]
I agree, having it in a nav drawer would be great. I think I started an attempt at creating a nav drawer for the app, but hit some bugs that I couldn't resolve (see #251 )

**Comment by misaochan** [comment]
Now that we have a nav drawer, what do you guys think about implementing this?

**Comment by misaochan** [comment]
@nicolas-raoul , do you know if there is an API for this?

**Comment by nicolas-raoul** [comment]
I think it is just a normal category. I am sure there is an API to get all media of a category.

**Comment by misaochan** [comment]
Ah okay, thanks!

**Comment by misaochan** [comment]
This will be implemented in the later half of the IEG renewal. The scope is still somewhat vague, so we need to discuss how the UI will look like, what exactly to display, etc.

**Use case:**
We plan on adding a "featured images" activity to the app, accessible via the navigation drawer. This will allow users to browse featured images on Commons, which aims to educate newcomers on what makes a good Commons picture, and inspire and motivate users to take great pictures in hopes of seeing their own picture in that section.

**Scope:**
- Look for API that fetches media from the featured pictures category - https://commons.wikimedia.org/wiki/Category:Featured_pictures_on_Wikimedia_Commons . How many should we fetch?
- Download thumbnails of images from API along with associated information
- Display thumbnails in UI, possibly along with title, description, author etc. What sort of view would we want to use for this? Should it be similar to the Contributions view?
- Link to this activity in the nav drawer

**Comment by neslihanturan** [comment]
> Display thumbnails in UI, possibly along with title, description, author etc. What sort of view would we want to use for this? Should it be similar to the Contributions view?

+1 for Contributions view

**Comment by neslihanturan** [comment]
Since I really liked this idea, maybe we can add it on third tab along with contributions, nearby, instead of another item on nav drawer.

**Comment by neslihanturan** [comment]
While working on mockups of this feature, some question come up:
1- Which information do we want to emphasize on grid layout? Only thumbnail, file name + thumbnail etc.
2- Which information do we want to emphasize on media detail page?

**Comment by misaochan** [comment]
I would personally not be in favour of 3 tabs - I think 2 is enough, too many would distract the user from our main focus. IMO this would be better as an item in the nav drawer instead of a tab.

1. It depends on what kind of layout we use - if it was similar to the current Contributions, then having a filename would be good I think

2. Off the top of my head - filename, desc, categories.

**Comment by neslihanturan** [comment]
Here are three suggestion of me for featured pictures. I would go with 3rd one since we want to encourage users to take better pictures to be in this list one day, displaying user names on first sight could help our purpose.
![mockupsfeatured](https://user-images.githubusercontent.com/3127881/32609360-7d357bee-c570-11e7-9db5-54f7725c78d6.png)

Besides since I don't want any license violations, here are attributions from top left corner:
- Original picture: [Uwe kils](https://commons.wikimedia.org/wiki/User:Uwe_kils), modifier [Amada44](https://commons.wikimedia.org/wiki/User:Amada44)
- NASA/JPL-Caltech/Univ. of Arizona (Actually this doesn't require attribution)
- [EduardoMSNeves](https://commons.wikimedia.org/w/index.php?title=User:EduardoMSNeves&action=edit&redlink=1)
- [The Photographer](https://commons.wikimedia.org/wiki/User:The_Photographer)
- © Tomas Castelazo, www.tomascastelazo.com / Wikimedia Commons / CC BY-SA 4.0
- Wiedehopf20
- [Petar Milošević](https://commons.wikimedia.org/wiki/User:PetarM)
- © Tomas Castelazo, www.tomascastelazo.com / Wikimedia Commons / CC BY-SA 4.0

**Comment by misaochan** [comment]
Looks good to me! I would be good with Option 3, but I think we need to find a way to show more clearly that the bottom text is the author. Perhaps a more significant difference in font between the top and bottom text?

**Comment by nicolas-raoul** [comment]
Even if visually option 1 is best, I prefer option 3 for our purpose.
How about showing "Uploaded by Uwe kils" instead of just "Uwe kils"? WIth users' sometimes weird usernames it is worth the precision.

Would tapping a picture show its details view with description/categories? Featured images always have proper categories, which could serve as a good example to follow.

**Comment by neslihanturan** [comment]
Actually I tried to fit categories to this page too, but there is no enough space. I agree adding Uploaded by indicator. And yes, I will send some media details sketch today.

**Comment by nicolas-raoul** [comment]
How about reusing the existing media details view, with just the added
field "author"?

**Comment by neslihanturan** [comment]
I remember some discussions about existing media details view here #873 . So maybe we need to redesign it for both featured and contributions view. What do you think @nicolas-raoul ?

**Comment by nicolas-raoul** [comment]
Yes the two will probably have much in common.

**Comment by misaochan** [comment]
I definitely wouldn't try and fit categories into the grid layout, haha. It would be way too cramped IMO.

I also agree with adding "Uploaded by", and using the same media details view UI for contributions and featured. However, if we do add the ability for users to modify their desc etc inside the app, that needs to be restricted solely to contributions (no concrete plans for that yet until 2018 though).

**Comment by neslihanturan** [comment]
So should we leave UI improvements for media detail to 2018 too? This would make me very happy though because I am having hard times to choose a design currently:)

**Comment by ujjwalagrawal17** [comment]
I did some research on finding list of images in a particular category. But was unable to find that API. instead we can use [API:Images](https://www.mediawiki.org/w/api.php?action=help&modules=query%2Bimages) to find all images in a particular page. 

Also I think most of the images from [Category:Featured_pictures_on_Wikimedia_Commons](https://commons.wikimedia.org/wiki/Category:Featured_pictures_on_Wikimedia_Commons)
will be available on [Commons:Featured_pictures page](https://commons.wikimedia.org/wiki/Commons:Featured_pictures)

**Comment by ujjwalagrawal17** [comment]
Can I work on this issue?

**Comment by neslihanturan** [comment]
Hi @ujjwalagrawal17 , did you noticed the UI is already ready on another branch. And the rest is in our IEG plans. Besides, if you feel ready to start a non-beginner friendly task please go for it:) I will assign it to you.

**Comment by misaochan** [comment]
Hi @ujjwalagrawal17 , sorry, we would like to be able to assign this issue to you, but it is already in our IEG plans for the next few months. Please do take a look at one of our other issues, thanks. :)
